### ISSUE TITLE
FTDI port

### ISSUE THREAD

**Issue by nbeAnomes** [OpenLightingProject/ola#1219]
Hello,

I would like to know why FTDI devices output port begins by 1, and not by 0 as others device.
From what I see, in the loop of the StartHook() function of the FtdiDmxDevice when the FtdiDmxOutputPort is created. Its port index is set with an offset of 1.

Anyway, thanks for your great work !

**Comment by peternewman** [comment]
That's a fair point, feel free to open a PR. :)

The port ID is generally less important than the Description, although it looks like the current code names them all the same, which isn't very helpful:
https://github.com/OpenLightingProject/ola/blob/master/plugins/ftdidmx/FtdiDmxPort.h#L62
https://github.com/OpenLightingProject/ola/blob/master/plugins/ftdidmx/FtdiWidget.h#L182
https://github.com/OpenLightingProject/ola/blob/master/plugins/ftdidmx/FtdiWidget.h#L148

See the code here for the sort of thing we can do, although I suspect given different models of FTDI based unit, we probably want to keep it fairly generic:
https://github.com/OpenLightingProject/ola/pull/1072/files#diff-004f973378d58eab249e16bc9a237804R74

I've also been trying to move the info like the serial number that's common to the device, rather than the port, to the device description, rather than the port description. See for example the changes here:
https://github.com/OpenLightingProject/ola/pull/1039/files#diff-520883e6359feb30989879075897b7f0R66

So if you're doing some updates, doing that at the same time too would be great.

Have you got a multi-port FTDI based device then?

**Comment by peternewman** [comment]
See my related issue regarding naming here https://github.com/OpenLightingProject/ola/issues/975

**Comment by peternewman** [comment]
@Keeper-of-the-Keys I think this is what you were referring to regarding FTDI port naming?

**Comment by Keeper-of-the-Keys** [comment]
@peternewman  Not quite, I have FTDI 2 boards with 4 ports, now when I create/assign a universe I can only chose which board(s) it goes to but not which of its' specific ports, they will get assigned in-order and if I for some reason did not assign the universes in-order or I named them instead of giving them numbers then currently it actually doesn't show anywhere (iirc the boards are at home so I can't check) to which specific port the universe is associated.

I have asked on the libftdi mailing list if there is a way to make the I/O LEDs of ports blink to "identify" ports, no answer on that yet (I tried something minor with that so far).

Is there currently a facility in OLA that the plugin just needs to export the port names/numbers properly or if this also not supported for any other devices?

**Comment by Keeper-of-the-Keys** [comment]
(So for instance I can't create a universe associated with port B before having created one associated with port A)

**Comment by peternewman** [comment]
Ah, that sounds like AllowMultiPortPatching() is set wrong:
http://docs.openlighting.org/ola/doc/latest/classola_1_1_device.html#aa2e036c37dea36394570618613368f08

And indeed the code supports that.

Lets get #1422 sorted and merged in first, then look at that and the other FTDI port naming/patching issues, but something like this should do the trick:
https://github.com/OpenLightingProject/ola/pull/1423/files

**Comment by Keeper-of-the-Keys** [comment]
@peternewman Very likely, the plugin was originally created with 1 port in mind and I can't say my expansion to supporting multiple ports was aware of all the intricacies of OLA, I agree lets get libftdi1 out there first and then move on to features :)

**Comment by peternewman** [comment]
See also (for historic reference) https://github.com/OpenLightingProject/ola/pull/394

**Comment by peternewman** [comment]
Hi @Keeper-of-the-Keys ,

Please see and test https://github.com/OpenLightingProject/ola/pull/1435 . If you're happy we'll merge, and that should fix your specific request here.

**Comment by peternewman** [comment]
So in terms of unique naming of individual ports @Keeper-of-the-Keys , see my comments here:
https://github.com/OpenLightingProject/ola/issues/1219#issuecomment-294134338

Essentially adding an i into the port's description should solve the issue. But we should probably remove a lot of the generic fluff into the device's description instead.
